### PR TITLE
feat(js-connectors): interactive smoke test for JS connectors

### PR DESCRIPTION
**Pull request by jkomyno**
`smoke-test-js` provides a handy tool for smoke testing our JS connectors implementations.
For the time being, is allows fetch a table whose columns are of several different types and printing the results.
For the moment, this is enough for our internal devs as it's useful to ensure:
- that the connection to PlanetScale is successful
- that submitting queries to the library Query Engine indeed calls the JS connector (when the `provider` name starts with `@prisma/`)
- that the results of those queries (which are JS-encoded) are properly understood by our Rust implementation, converted to Quaint, and later serialised back to JSON). 

Refer to the README.md in `./js-connectors/smoke-test-js/` for detailed info and docs. Merge https://github.com/prisma/prisma-engines/pull/4080 (shipping the data-type support for PlanetScale) before testing this PR out.

This PR closes https://github.com/prisma/team-orm/issues/249.

### CODE REVIEW

File: query-engine/js-connectors/smoke-test-js/README.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+- Create a new database on [PlanetScale](https://planetscale.com/)
+- Go to `Settings` > `Passwords`, and create a new password for the `main` database branch. Select the `Prisma` template and copy the generated URL (comprising username, password, etc). Paste it in the `JS_PLANETSCALE_DATABASE_URL` environment variable in `.envrc`.
+- Create a new `shadow` database branch. Repeat the steps above (selecting the `shadow` branch instead of `main`), and paste the generated URL in the `JS_PLANETSCALE_SHADOW_DATABASE_URL` environment variable in `.envrc`.
+
+In the current directory:
+- Run `prisma migrate reset` to populate the remote PlanetScale database with the "smoke test" data.
+- Change the `provider` name in [./prisma/schema.prisma](./prisma/schema.prisma) from `mysql` to `@prisma/planetscale`.
+
+Note: you used to be able to run these Prisma commands without changing the provider name, but [#4074](https://github.com/prisma/prisma-engines/pull/4074) broke that (see https://github.com/prisma/prisma-engines/pull/4074/files#r1273323391).
```
Comment (jkomyno) [inline query-engine/js-connectors/smoke-test-js/README.md:26]:
```suggestion
Note: you used to be able to run these Prisma commands without changing the provider name, but [#4074](https://github.com/prisma/prisma-engines/pull/4074) changed that (see https://github.com/prisma/prisma-engines/pull/4074#issuecomment-1649942475).
```

File: query-engine/js-connectors/smoke-test-js/prisma/migrations/planetscale/migration.sql [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    varbinary_column_null VARBINARY(128),
+    --
+    -- ColumnType::VarBinary, PlanetScale::{VARBINARY}
+    blob_column BLOB NOT NULL,
+    blob_null BLOB,
+    --
+    -- ColumnType::Set, PlanetScale::{SET}
+    set_column SET('option1', 'option2', 'option3') NOT NULL,
+    set_column_null SET('option1', 'option2', 'option3')
+);
```
Comment (miguelff) [inline query-engine/js-connectors/smoke-test-js/prisma/migrations/planetscale/migration.sql:91]: super!

File: query-engine/js-connectors/smoke-test-js/prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  value1
+  value2
+  value3
+}
+
+enum type_test_enum_column_null {
+  value1
+  value2
+  value3
+}
```
Comment (miguelff) [inline query-engine/js-connectors/smoke-test-js/prisma/schema.prisma:80]: Nice, we can extend this in the future for queries with relations!

**Review by miguelff — APPROVED** [review]
Lets ship this and move to having a vertical slice of node drivers for PS.

**Pull request merged**